server/src/main/java/com/thoughtworks/go/config/CachedGoConfig.java

```diff
@@ -139,12 +136,8 @@ public synchronized ConfigSaveState writeFullConfigWithLock(FullConfigUpdateComm
     }
 
     public synchronized void upgradeConfig() throws Exception {
-        if(systemEnvironment.optimizeFullConfigSave()) {
-            GoConfigHolder goConfigHolder = goConfigMigrator.migrate();
-            saveValidConfigToCacheAndNotifyConfigChangeListeners(goConfigHolder);
-        } else {
-            dataSource.upgradeIfNecessary();
```

Comment:
Did you get a chance to look at the `GoConfigMigration#upgradeIfNecessary()`, can this method be removed now?